tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/eval/ParseEvaluation.scala

```diff
@@ -0,0 +1,148 @@
+
+    (parseSource.parseIterator map { candidateParse =>
+      analyzer(candidateParse)
+    }) reduce { (x, y) =>
+      (x.keySet ++ y.keySet map { key =>
+        (key, x.getOrElse(key, 0.0) + y.getOrElse(key, 0.0))
+      }).toMap
+    }
+  }
+
+  /** Runs a standard set of scoring functions on a set of candidate parses with respect to a
+    * bank of gold parses.
+    *
+    * @param candidateSource the source of candidate parses
+    * @param goldParseBank a bank containing the gold parses
+    */
+  def performStandardEvaluation(
+    candidateSource: PolytreeParseSource,
+    goldParseBank: ParseBank
+  ): Unit = {
+
+    val goldScoringFunctions: Seq[ParseScore] = Seq(
+      UnlabeledAttachmentScore(goldParseBank),
+      LabeledAttachmentScore(goldParseBank),
+      UnlabeledLostTokens(goldParseBank),
+      LabeledLostTokens(goldParseBank),
+      PostagAccuracy(goldParseBank)
+    )
+    val allScores = for {
+      scoringFunction <- goldScoringFunctions
+    } yield {
+      (scoringFunction.name, ParseEvaluation.scoreParseSource(scoringFunction, candidateSource))
+    }
+    allScores foreach {
+      case (scoreName, (userNumer, userDenom)) =>
+        println(f"$scoreName: ${100 * userNumer.toDouble / userDenom}%.2f" + "%")
+    }
+  }
+
+  /** Runs a standard set of analyzers on a set of candidate parses with respect to a
+    * bank of gold parses.
+    *
+    * @param candidateSource the source of candidate parses
+    * @param goldParseBank a bank containing the gold parses
+    */
+  def performStandardAnalysis(
+    candidateSource: PolytreeParseSource,
+    goldParseBank: ParseBank
+  ): Unit = {
+
+    val analyzers: Seq[ParseAnalyzer] = Seq(
+      MisattachmentAnalyzer(goldParseBank, ignoreLabel = true),
+      LostTokensAnalyzer(goldParseBank),
+      CposErrorAnalyzer(goldParseBank)
+    )
+    val allAnalyses = for {
+      analyzer <- analyzers
+    } yield {
+      (analyzer.name, ParseEvaluation.analyzeParseSource(analyzer, candidateSource))
+    }
+    allAnalyses foreach {
+      case (analyzerName, analysis) =>
+        println(s"ARCLABEL\t$analyzerName")
+        analysis foreach {
+          case (label, count) =>
+            println(s"${label.toLowerCase}\t$count")
+        }
+    }
+  }
+
+  /** Runs a standard contrastive evaluation of a set of candidate parsers ("users")
+    * with respect to a bank of gold parses.
+    *
+    * @param candidateSources the map from user names to candidate parses
+    * @param goldParseBank a bank containing the gold parses
+    */
+  def performStandardMultiUserEvaluation(
+    candidateSources: Map[String, PolytreeParseSource],
+    goldParseBank: ParseBank
+  ): Unit = {
+
+    val goldScoringFunctions: Seq[ParseScore] = Seq(
+      UnlabeledAttachmentScore(goldParseBank),
+      LabeledAttachmentScore(goldParseBank),
+      UnlabeledLostTokens(goldParseBank),
+      LabeledLostTokens(goldParseBank),
+      PostagAccuracy(goldParseBank)
```

Comment:
As below, that sounds wise, but I'll defer that to the next PR, to see what kind of refactoring is best once all my analysis code is in.